### PR TITLE
Add buildkite.yml.

### PR DESCRIPTION
**Pull request by jb-mackay**
Adds a buildkite pipeline for ClimaCoupler.

Executes the sea breeze integration test

### CODE REVIEW

File: bors.toml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 status = [
+  "buildkite/climacoupler-ci
```
Comment (charleskawczynski) [inline bors.toml:2]:
```suggestion
  "buildkite/climacoupler-ci“,
```

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+env:
+  JULIA_VERSION: "1.7.2"
+  OPENMPI_VERSION: "4.0.4"
+  CUDA_VERSION: "11.2"
+  OPENBLAS_NUM_THREADS: 1
+  BUILDKITE_COMMIT: "${BUILDKITE_COMMIT}"
+  BUILDKITE_BRANCH: "${BUILDKITE_BRANCH}"
+  JULIA_DEPOT_PATH: "${BUILDKITE_BUILD_PATH}/${BUILDKITE_PIPELINE_SLUG}/depot/cpu"
```
Comment (charleskawczynski) [inline .buildkite/pipeline.yml:8]:
```suggestion
```

Let's skip out on using the Julia depot until we get its consequences documented somewhere.

**Review by charleskawczynski — APPROVED** [review]
Looks good!

**Comment by jb-mackay** [comment]
> Looks good!

Thanks @charleskawczynski! You're video made this super easy.

**Pull request merged**